src/step_definitions/programManagementSteps.js

```diff
@@ -775,10 +777,12 @@ When(
     if (args1.includes("*")) {
       programName = program.Name;
     } else programName = args1;
-    await page.click({
+    selector = {
       selector: `//*[@id='app']//main//a[normalize-space(.)='${programName}']`,
       locateStrategy: "xpath",
-    });
+    };
+    await page.waitForElementVisible(selector);
+    await page.click(selector);
```

Comment:
If this pattern makes TAF runs more reliable/deterministic, we should use it everywhere going forward.